### PR TITLE
Fix potential deadlock regarding buffer condvar

### PR DESCRIPTION
**Pull request by skyline75489**

### CODE REVIEW

**Comment by skyline75489** [comment]
Repro in https://github.com/alacritty/alacritty/issues/5266

File: src/lib.rs [hunk 1]
```diff
@@ -166,7 +166,11 @@ impl io::Read for EventedAnonRead {
             // We avoid the race by re-checking the queue is empty like this, and undo the
             // readiness setting if necessary.
             if !self.consumer.is_empty() {
-                self.inner.readiness.set_readiness(Ready::empty())?;
+                self.inner.readiness.set_readiness(Ready::readable())?;
```
Comment (davidhewitt) [inline src/lib.rs:169]: This was definitely a bug, good spot!

**Review by davidhewitt — APPROVED** [review]
👍 thanks, I'm going to do some quick follow-up fixes inpired by this; I'll ping you for review if that's ok!

**Comment by chrisduerr** [comment]
I'd love a ping when you draft a new release. Happy to look at other bugs for reviews but can't test so that's probably rather useless.

**Pull request merged**